packages/rule-parser/README.md

```diff
@@ -0,0 +1,163 @@
+# Rule Parser
+
+This package provides a rule parser to evaluate logical rules into a boolean value.
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/rule-parser --save
+```
+
+## API
+
+> [!NOTE]
+> This package is only meant to be used by implementors, not direct developers. implementors are then responsible for collecting rules from developers or users, either via UI (like Blocks Visibility), or code (like DataForms), evalute the rules against their provided context, and use the outcome however they see fit.
+
+The package accepts rules in the shape of `[ source, operator, target ]` and can be infinitely nested on a variation of combinators, that resemble `||` (OR, ANY) or `&&` (AND, ALL).
+
+### Examples of rules
+
+1. Simple rule
+
+```js
+const rule = [ [ 'user.role', 'is', 'editor' ] ];
+```
+
+2. Array of rules
+
+```js
+const rules = [
+	[ 'user.role', 'is', 'editor' ],
+	[ 'post.categories', 'contain', 'tutorials' ],
+];
+```
+
+3. Array of rules with explicit combinator. If no combinator is set, `ALL` will be used.
+
+```js
+const rules = [
+	'ALL',
+	[
+		[ 'user.role', 'is', 'editor' ],
+		[ 'post.categories', 'contain', 'tutorials' ],
+	],
+];
+```
+
+4. Nested arrays
+
+```JS
+const rules = [
+    'ANY',
+    [
+        [ 'user.role', 'is', 'editor' ],
+        [ 'post.categories', 'contain', 'tutorials' ],
+        [
+            'ALL',
+            [
+                [ 'user.id', 'in', [ 1, 2, 3 ] ],
```

Comment:
The very initial implementation we had at Checkout operated that way, this added the need to parse things. I think we might want to provide even greater flexibility to consumers, so instead of passing an object of key value, they can also pass an object of keys and functions/loaders, or receive a single loader function and they decide what to do with it.

For now we can probably settle on supporting dot notation out of the box?